third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -1116,19 +1111,19 @@ where
         traversal_context: &mut TraversalContext,
         addr: AccountAddress,
         ty: &Type,
-    ) -> PartialVMResult<&'c mut GlobalValue> {
-        if !data_cache.contains_resource(&addr, ty) {
-            let entry = self.create_and_charge_data_cache_entry(
-                resource_resolver,
+    ) -> PartialVMResult<bool> {
+        if !data_cache.contains_resource_existence(&addr, ty) {
+            data_cache.create_and_insert_or_upgrade_and_charge_data_cache_entry(
                 module_storage,
-                gas_meter,
+                resource_resolver,
+                Some(gas_meter),
                 traversal_context,
-                addr,
+                &addr,
                 ty,
+                false,
```

Comment:
let's add a one liner that if existence optimization is not enabled, this will still be overridden by data cache.